content/doc/developer/internationalization/new-internationalization.adoc

```diff
@@ -0,0 +1,339 @@
+---
+title: New Internationalization
+layout: developersection
+---
+ 
+https://wiki.jenkins.io/display/JENKINS/How+to+view+Jenkins+in+your+language[How
+to view Jenkins in your language].
+
+'''''
+
+== Translations to Specific languages
+
+* https://wiki.jenkins.io/display/JENKINS/Chinese+%28Traditional%29+Translation[Chinese
+(Traditional) Translation]
+* https://wiki.jenkins.io/display/JENKINS/Chinese+%28Simplified%29+Translation[Chinese
+(Simplified) Translation] 
+* https://wiki.jenkins.io/display/JENKINS/French+Localization[French
+Localization]
+* https://wiki.jenkins.io/display/JA/Japanese+Localization[Japanese
+Localization]
+* https://wiki.jenkins.io/display/JENKINS/Polish+Localization[Polish
+Localization]
+* https://wiki.jenkins.io/display/JENKINS/Spanish+Translation[Spanish
+Translation]
+* https://wiki.jenkins.io/display/JENKINS/Brazilian+Portuguese+Translation[Brazilian
+Portuguese Translation]
+
+== What developers need to know/do
+
+Jenkins' i18n support has some aspects that developers need to be aware
+of.
+
+. Generation of type-safe `+Messages+` classes from
+`+Messages.properties.+`
+. Marking messages in jelly files.
+. Creation of html files for help tips.
+
+=== Generation of `+Messages+` classes
+
+Jenkins uses http://java.net/projects/localizer/[localizer] to generate
+classes for accessing message resources in a type-safe way. For each
+`+src/main/resources/**/Messages.properties+`, a Messages class is
+generated. See the referenced page for how this class looks like. If
+your IDE fails to find these classes, manually add
+"target/generated-sources/localizer" directory to your source root.
+
+Wherever the code returns a String for display purpose (such as
+`+Descriptor.getDisplayName()+`), use the Messages class to obtain a
+localized message. At the runtime, a proper locale is selected
+automatically. A typical workflow for this is as follows:
+
+. You identify messages that need be localized
+. You place that in Messages.properties. One can choose to have this
+file for each package, or you could just have one such file for the
+whole module/plugin.
+. You run `+mvn compile+` once to re-generate `+Messages.java+`
+. Update your code to use the newly generated message formatting method
++
+[cols="",]
+|===
+|As usual, looking at how the core code does this might help you get the
+idea of how to do this.
+|===
++
+[cols="",]
+|===
+|*Note:* If your message contains single quote character ('), then you
+need to escape it with another single quote character (i.e. ''). So
```

Comment:
```suggestion
need to escape it with another single quote character such as ''. So
```
Need to check how that '' (two single quote characters) is rendering.Github is not rendering it correctly, even in this comment!